packages/core/frontend/sirius-components-core/src/onboarding/NewRepresentationArea.tsx

```diff
@@ -0,0 +1,157 @@
+
+const useNewRepresentationAreaStyles = makeStyles(() => ({
+  card: {
+    backgroundColor: 'transparent',
+    maxWidth: '330px',
+    minWidth: '330px',
+  },
+  subtitles: {
+    textOverflow: 'ellipsis " [..]";',
+  },
+  cardContent: {
+    height: '100%',
+    maxHeight: '400px',
+    overflowY: 'auto',
+  },
+}));
+
+const createRepresentationMutation = gql`
+  mutation createRepresentation($input: CreateRepresentationInput!) {
+    createRepresentation(input: $input) {
+      __typename
+      ... on CreateRepresentationSuccessPayload {
+        representation {
+          id
+          label
+          kind
+          __typename
+        }
+      }
+      ... on ErrorPayload {
+        message
+      }
+    }
+  }
+`;
+
+export const NewRepresentationArea = ({
+  editingContextId,
+  representationDescriptions,
+  selection,
+  setSelection,
+  readOnly,
+}: NewRepresentationAreaProps) => {
+  const [message, setMessage] = useState('');
+  const [open, setOpen] = useState(false);
+  const classes = useNewRepresentationAreaStyles();
+  const selectedItem = selection.entries.length > 0 ? selection.entries[0] : null;
+
+  // Representation creation
+  const [createRepresentation, { loading, data, error }] = useMutation(createRepresentationMutation);
+  useEffect(() => {
+    if (!loading && !error && data?.createRepresentation) {
+      const { createRepresentation } = data;
+      if (createRepresentation.representation) {
+        const { id, label, kind } = createRepresentation.representation;
+        setSelection({ entries: [{ id, label, kind }] });
+      } else if (createRepresentation.__typename === 'ErrorPayload') {
+        setOpen(true);
+        setMessage(createRepresentation.message);
+      }
+    }
+  }, [loading, data, error, setSelection]);
+  const onCreateRepresentation = (representationDescriptionId) => {
+    const selected = representationDescriptions.find((candidate) => candidate.id === representationDescriptionId);
+    const objectId = selectedItem.id;
+    const input = {
+      id: uuid(),
+      editingContextId,
+      objectId,
+      representationDescriptionId,
+      representationName: selected.defaultName,
+    };
+    createRepresentation({ variables: { input } });
+  };
+
+  // Representation Descriptions list
+  let newRepresentationList = representationDescriptions.map((representationDescription) => {
+    return (
+      <ListItem
+        disableGutters
+        button
+        key={representationDescription.id}
+        data-testid={representationDescription.id}
+        onClick={() => {
+          onCreateRepresentation(representationDescription.id);
+        }}>
+        <ListItemIcon>
+          <Collections htmlColor="Black" fontSize="small" />
+        </ListItemIcon>
+        <ListItemText primary={representationDescription.defaultName} />
+      </ListItem>
+    );
+  });
+
+  let title = 'Create a new Representation';
```

Comment:
- Use `const` instead of `let` (pretty much everywhere in this file)
- Inline this in the JSX